lib/semian/lru_hash.rb

```diff
@@ -0,0 +1,53 @@
+class LRUHash
+  extend Forwardable
+  def_delegators :@table, :size, :count, :empty?, :keys, :clear, :delete
+
+  attr_reader :table, :minimum_time_in_lru
+  def initialize(minimum_time_in_lru = 300)
+    @minimum_time_in_lru = minimum_time_in_lru
+    @table = {}
+    @lock = Mutex.new
```

Comment:
You should pass the `thread_safety_disabled` and replace `Mutex` with a mock here, e.g. `class NoopMutex; def synchronize(*); end; end` when we don't have thread-safety enabled. A mutex is a substantial performance hit, so let's not use it unless necessary.